compiler/hash-typecheck/src/new/passes/scope_discovery.rs

```diff
@@ -565,4 +619,79 @@ impl<'tc> ast::AstVisitor for ScopeDiscoveryPass<'tc> {
             .add_error(TcError::TraitsNotSupported { trait_location: self.node_location(node) });
         Ok(())
     }
+
+    type FnDefRet = ();
+    fn visit_fn_def(&self, node: AstNodeRef<ast::FnDef>) -> Result<Self::FnDefRet, Self::Error> {
+        // Get the function name from the name hint.
+        let fn_def_name = self.take_name_hint_or_create_internal_name();
+
+        // Create a function definition
+        let fn_def_id = self.fn_ops().create_fn_def(FnDefData {
+            name: fn_def_name,
+            body: FnBody::Defined(self.new_term_hole()),
+            ty: FnTy {
+                implicit: false,
+                is_unsafe: false,
+                params: self.create_hole_params(&node.params),
+                pure: false,
+                return_type: self.new_ty_hole(),
+            },
+        });
```

Comment:
When does this deal closures?

Reply:
It already does, though what is still to be done is to have info on fndefs about what variables they capture

Reply:
This will happen in the symbol resolution stage

Reply:
But how can you determine if a function definition is a function or a closure?